contrib/lua/premake5.lua

```diff
@@ -21,8 +21,8 @@ project "lua-lib"
 		"etc/*.c"
 	}
 
-	configuration "linux or bsd or hurd or aix"
+	filter "system:linux or system:bsd or system:hurd or system:aix"
```

Comment:
It's like I say above, I think this is one of premake's central core mechanic's, so it's definitely worthy of special attention. We could spin off a separate thread if you like. I'd ask that, at least in the mean time, you don't write code like `not scope:thing`, and we can decide what sort of expressions we want to support on this elsewhere?